plugin-api/src/main/java/org/hyperledger/besu/plugin/BesuPlugin.java

```diff
@@ -36,6 +38,16 @@
     return Optional.of(this.getClass().getName());
   }
 
+  /**
+   * Called when the plugin is loaded so that you can register cli options to be parsed
+   *
+   * @param cliOptions the hook to register PicoCLIOptions.
+   */
+  default void registerPicoCLIOptions(final PicoCLIOptions cliOptions) {
```

Comment:
My concern is that, even with a default interface method, it forces all plugins to be PicoCLI aware and deeply embedded in the lifecycle.  Not every plugin cares or should care about PicoCLI, which is why I exposed it as a service.

Reply:
Yeah, I'm not a fan of this either. However, it is more explicit than the other proposal https://github.com/hyperledger/besu/pull/2768.